rasa_nlu/utils/md_to_rasa.py

```diff
@@ -0,0 +1,98 @@
+import codecs
+import json
+import sys
+import re
+import io
+from rasa_nlu.training_data import Message
+
+ent_regex = re.compile('\[(?P<value>[^\]]+)]\((?P<entity>[^:)]+)\)')  # [restaurant](what)
+ent_regex_with_value = re.compile('\[(?P<synonym>[^\]]+)\]\((?P<entity>\w*?):(?P<value>[^)]+)\)')  # [open](open:1)
+intent_regex = re.compile('##\s*intent:(.+)')
+synonym_regex = re.compile('##\s*synonym:(.+)')
+example_regex = re.compile('\s*-\s*(.+)')
+
+
+class MarkdownToRasa:
+    """ Converts training examples written in markdown to standard rasa json format """
+    def __init__(self, file_name):
+        self.file_name = file_name
+        self.current_intent = None  # set when parsing examples from a given intent
+        self.current_word = None  # set when parsing synonyms
+        self.common_examples = []
+        self.entity_synonyms = {}
+        self.load()
+
+    def get_example(self, utter):
+        example = {'intent': self.current_intent}
+        entities = []
+        example['text'] = utter
+        for regex in [ent_regex, ent_regex_with_value]:
+            example['text'] = re.sub(regex, r"\1", example['text'])
+            ent_matches = re.finditer(regex, utter)
+            for matchNum, match in enumerate(ent_matches):
+                entity_value_in_utter = match.groupdict()['synonym'] if 'synonym' in match.groupdict() \
+                    else match.groupdict()['value']
+                entities.append({
+                    'entity': match.groupdict()['entity'],
+                    'value': match.groupdict()['value'],
+                    'start': example['text'].index(entity_value_in_utter),
+                    'end': (example['text'].index(entity_value_in_utter) + len(entity_value_in_utter))
+                })
+        if len(entities) > 0:
```

Comment:
I think 
```python
return Message(utter, {'intent': self.current_intent, 'entities': entities})
```
should be fine (instead of if and L44) (also removes the need for the example variable)